server/src/logic/sync/domain/use-cases/sync-all-folders-data.use-case.ts

```diff
@@ -66,7 +67,32 @@ export class SyncAllFolders implements SyncAllFoldersUseCase {
   }
 
   processItems(syncEntity: SyncEntity, items: SyncItem[]) {
+    const filteredMimetypes = (syncEntity.filters?.fileExtensions?.extensions || '')
+      .split(',')
+      .map((ext) => lookup(ext.trim()) || '')
+      .filter((ext) => ext);
     return of(...items).pipe(
+      filter((item) => {
+        let isExtensionOk = true;
+        if (!item.mimeType) {
+          isExtensionOk = false;
+        } else {
+          if (filteredMimetypes.length > 0) {
+            const isFiltered = filteredMimetypes.includes(item.mimeType);
+            isExtensionOk = syncEntity.filters?.fileExtensions?.exclude ? !isFiltered : isFiltered;
+          }
+        }
+        let isDateOk = true;
+        if (item.modifiedGMT && syncEntity.filters?.modified) {
+          if (syncEntity.filters.modified.from) {
+            isDateOk = item.modifiedGMT >= syncEntity.filters.modified.from;
+          }
+          if (syncEntity.filters.modified.to) {
```

Comment:
I noticed that the date filtering is not working in specific cases.
for instance: if modifiedGTM is lesser than "to", then the "from" filter is not taken into account.